syntax = "proto3";
import "google/protobuf/timestamp.proto";

service categoryExchange {
  //request info from logger like IP,category to filter
  rpc getInformation (infoRequest) returns (infoReply) {}
  //send\get events
  rpc events (evtMgr) returns (getEvents) {}
}

message infoRequest{
  bool information =1;
}

message infoReply{
  string category=1;
  int32 logger_IP=2;
}

message evtMgr {
  int64 id = 1;
  google.protobuf.Timestamp time = 2;  // Unique ID number for this person.
  int32 type = 3;
  string src = 4;
  int32 cat = 5;
  repeated string dataList = 6; //data content

  string hostname = 7;
  string username = 8;
  string os = 9;
}

message getEvents{
}
//protoc --proto_path=ProtoBuf --python_out=./ ProtoBuf/evtmanager.proto
//Added to github